File: content-service/src/main/java/com/smatech/playlizt/content/entity/Content.java
package com.smatech.playlizt.content.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "content", indexes = {
    @Index(name = "idx_content_creator", columnList = "creator_id"),
    @Index(name = "idx_content_category", columnList = "category"),
    @Index(name = "idx_content_published", columnList = "is_published")
})
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Content {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, name = "creator_id")
    private Long creatorId;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false, length = 100)
    private String category;

    @ElementCollection(fetch = FetchType.EAGER)
    @CollectionTable(name = "content_tags", joinColumns = @JoinColumn(name = "content_id"))
    @Column(name = "tag")
    @Builder.Default
    private List<String> tags = new ArrayList<>();

    @Column(name = "thumbnail_url", length = 500)
    private String thumbnailUrl;

    @Column(name = "video_url", length = 500)
    private String videoUrl;

    @Column(name = "duration_seconds")
    private Integer durationSeconds;

    @Column(name = "ai_generated_description", columnDefinition = "TEXT")
    private String aiGeneratedDescription;

    @Column(name = "ai_predicted_category", length = 100)
    private String aiPredictedCategory;

    @Column(name = "ai_relevance_score", precision = 3, scale = 2)
    private BigDecimal aiRelevanceScore;

    @Column(name = "ai_content_rating", length = 20)
    private String aiContentRating;

    @Column(name = "ai_sentiment", length = 20)
    private String aiSentiment;

    @CreationTimestamp
    @Column(nullable = false, updatable = false, name = "created_at")
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(nullable = false, name = "updated_at")
    private LocalDateTime updatedAt;

    @Column(nullable = false, name = "is_published")
    private Boolean isPublished = false;

    @Column(name = "view_count")
    private Long viewCount = 0L;
}

File: playlizt-frontend/playlizt_app/lib/models/content.dart
class Content {
  final int id;
  final int creatorId;
  final String title;
  final String? description;
  final String category;
  final List<String> tags;
  final String? thumbnailUrl;
  final String? videoUrl;
  final int? durationSeconds;
  final String? aiGeneratedDescription;
  final String? aiPredictedCategory;
  final double? aiRelevanceScore;
  final String? aiContentRating;
  final String? aiSentiment;
  final DateTime createdAt;
  final DateTime updatedAt;
  final bool isPublished;
  final int viewCount;
  final int? resumePositionSeconds;
  
  Content({
    required this.id,
    required this.creatorId,
    required this.title,
    this.description,
    required this.category,
    required this.tags,
    this.thumbnailUrl,
    this.videoUrl,
    this.durationSeconds,
    this.aiGeneratedDescription,
    this.aiPredictedCategory,
    this.aiRelevanceScore,
    this.aiContentRating,
    this.aiSentiment,
    required this.createdAt,
    required this.updatedAt,
    required this.isPublished,
    required this.viewCount,
    this.resumePositionSeconds,
  });
  
  factory Content.fromJson(Map<String, dynamic> json) {
    return Content(
      id: json['id'] ?? 0,
      creatorId: json['creatorId'] ?? 0,
      title: json['title'] ?? 'Untitled',
      description: json['description'],
      category: json['category'] ?? 'Uncategorized',
      tags: List<String>.from(json['tags'] ?? []),
      thumbnailUrl: json['thumbnailUrl'],
      videoUrl: json['videoUrl'],
      durationSeconds: json['durationSeconds'],
      aiGeneratedDescription: json['aiGeneratedDescription'],
      aiPredictedCategory: json['aiPredictedCategory'],
      aiRelevanceScore: json['aiRelevanceScore']?.toDouble(),
      aiContentRating: json['aiContentRating'],
      aiSentiment: json['aiSentiment'],
      createdAt: json['createdAt'] != null 
          ? DateTime.parse(json['createdAt']) 
          : DateTime.now(),
      updatedAt: json['updatedAt'] != null 
          ? DateTime.parse(json['updatedAt']) 
          : DateTime.now(),
      isPublished: json['isPublished'] ?? false,
      viewCount: json['viewCount'] ?? 0,
      resumePositionSeconds: json['resumePositionSeconds'],
    );
  }
  
  String get displayDescription => aiGeneratedDescription ?? description ?? 'No description available';
  
  String get formattedDuration {
    if (durationSeconds == null) return '';
    final duration = Duration(seconds: durationSeconds!);
    final hours = duration.inHours;
    final minutes = duration.inMinutes.remainder(60);
    final seconds = duration.inSeconds.remainder(60);
    
    if (hours > 0) {
      return '$hours:${minutes.toString().padLeft(2, '0')}:${seconds.toString().padLeft(2, '0')}';
    } else {
      return '$minutes:${seconds.toString().padLeft(2, '0')}';
    }
  }

  String get formattedResumePosition {
    if (resumePositionSeconds == null || resumePositionSeconds == 0) return '';
    final duration = Duration(seconds: resumePositionSeconds!);
    final hours = duration.inHours;
    final minutes = duration.inMinutes.remainder(60);
    final seconds = duration.inSeconds.remainder(60);

    if (hours > 0) {
      return '$hours:${minutes.toString().padLeft(2, '0')}:${seconds.toString().padLeft(2, '0')}';
    } else {
      return '$minutes:${seconds.toString().padLeft(2, '0')}';
    }
  }
}
